iree/tools/init_mlir_passes.h

```diff
@@ -49,63 +49,20 @@ namespace mlir {
   // https://reviews.llvm.org/D77322
-  ::mlir::registerPass(
-      "affine-loop-fusion", "Fuse affine loop nests",
-      []() -> std::unique_ptr<Pass> { return mlir::createLoopFusionPass(); });
-  ::mlir::registerPass("affine-pipeline-data-transfer",
-                       "Pipeline non-blocking data transfers between "
-                       "explicitly managed levels of the memory hierarchy",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createPipelineDataTransferPass();
-                       });
-  ::mlir::registerPass(
-      "cse", "Eliminate common sub-expressions",
-      []() -> std::unique_ptr<Pass> { return mlir::createCSEPass(); });
-  ::mlir::registerPass("canonicalize", "Canonicalize operations",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createCanonicalizerPass();
-                       });
-  ::mlir::registerPass(
-      "inline", "Inline function calls",
-      []() -> std::unique_ptr<Pass> { return mlir::createInlinerPass(); });
-  ::mlir::registerPass("snapshot-op-locations",
-                       "Generate new locations from the current IR",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createLocationSnapshotPass();
-                       });
-  ::mlir::registerPass(
-      "loop-coalescing",
-      "Coalesce nested loops with independent bounds into a single loop",
-      []() -> std::unique_ptr<Pass> {
-        return mlir::createLoopCoalescingPass();
-      });
-  ::mlir::registerPass("loop-invariant-code-motion",
-                       "Hoist loop invariant instructions outside of the loop",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createLoopInvariantCodeMotionPass();
-                       });
-  ::mlir::registerPass("memref-dataflow-opt",
-                       "Perform store/load forwarding for memrefs",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createMemRefDataFlowOptPass();
-                       });
-  ::mlir::registerPass(
-      "parallel-loop-collapsing",
-      "Collapse parallel loops to use less induction variables",
-      []() -> std::unique_ptr<Pass> {
-        return mlir::createParallelLoopCollapsingPass();
-      });
-  ::mlir::registerPass(
-      "print-op-stats", "Print statistics of operations",
-      []() -> std::unique_ptr<Pass> { return mlir::createPrintOpStatsPass(); });
-  ::mlir::registerPass("strip-debuginfo",
-                       "Strip debug info from all operations",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createStripDebugInfoPass();
-                       });
-  ::mlir::registerPass(
-      "symbol-dce", "Eliminate dead symbols",
-      []() -> std::unique_ptr<Pass> { return mlir::createSymbolDCEPass(); });
+  ::mlir::registerAffineLoopFusionPass();
```

Comment:
Hmm the comment and some of the other stuff here looks out of place. We should clean it up. Probably easier to do that after merging to main. OTOH this looks like fallout from weirdness in the LLVM integrate, so maybe we should fix it on the Google branch first.

Reply:
Got you. Let's merge this to main first, and fix the issue on google branch.